packages/engine/crawlers/solana/transactionHistory.ts

```diff
@@ -0,0 +1,530 @@
+
+	let amount = 0;
+
+	if (parsedInstructionInfo?.lamports) {
+		amount = parsedInstructionInfo?.lamports / 10 ** 9;
+	}
+
+	if (parsedInstructionInfo?.amount) {
+		amount = parsedInstructionInfo?.amount / 10 ** token.account.decimals;
+	}
+
+	if (parsedInstructionInfo?.tokenAmount?.amount) {
+		amount =
+			parseFloat(parsedInstructionInfo?.tokenAmount.amount) /
+			10 ** token.account.decimals;
+	}
+
+	if (token.account.mint === tokeForFee.account.mint) {
+		amount = parsedInstructionInfo?.amount / 10 ** token.account.decimals;
+	}
+
+	if (token.account.mint === solMint) {
+		amount = parsedInstructionInfo?.lamports / 10 ** 9;
+	}
+
+	if (token.metadata?.mpl) {
+		amount = 1;
+	}
+
+	return amount;
+};
+
+const getTransactionType = (
+	transaction: ParsedTransactionWithMeta,
+	ownerPublicKey: string,
+) => {
+	if (
+		!transaction.meta?.preTokenBalances ||
+		!transaction.meta?.postTokenBalances ||
+		transaction.meta.postTokenBalances.length === 0 ||
+		transaction.meta.preTokenBalances.length === 0
+	) {
+		const sender =
+			transaction.transaction.message.accountKeys[0].pubkey.toString();
+
+		const type = ownerPublicKey === sender ? 'sent' : 'received';
+
+		return type;
+	}
+
+	const { preBalance, postBalance } = getTransactionBalances(
+		transaction.meta,
+		ownerPublicKey,
+	);
+
+	const type = postBalance - preBalance < 0 ? 'sent' : 'received';
+
+	return type;
+};
+
+const getTransactionPartners = (
+	transaction: ParsedTransactionWithMeta,
+	ownerPublicKey: string,
+	type: 'sent' | 'received',
+) => {
+	let sender = '';
+	let receiver = '';
+
+	if (
+		!transaction.meta?.preTokenBalances ||
+		!transaction.meta?.postTokenBalances ||
+		transaction.meta.postTokenBalances.length === 0 ||
+		transaction.meta.preTokenBalances.length === 0
+	) {
+		const sender =
+			transaction.transaction.message.accountKeys[0].pubkey.toString();
+		const receiver =
+			transaction.transaction.message.accountKeys[1].pubkey.toString();
+
+		return {
+			sender,
+			receiver,
+		};
+	}
+
+	const partner = transaction.meta?.postTokenBalances.filter(
+		(item) => item.owner !== ownerPublicKey,
+	);
+
+	if (type === 'sent') {
+		sender = ownerPublicKey;
+		receiver = partner[0].owner ?? '';
+	} else {
+		sender = partner[0].owner ?? '';
+		receiver = ownerPublicKey;
+	}
+
+	return {
+		sender,
+		receiver,
+	};
+};
+
+export const getTransactionDetails = async (
+	solanaContext: SolanaContext,
+	parsedTransaction: ParsedTransactionWithMeta,
+	ownerPublicKey: string,
+) => {
+	const instructions = parsedTransaction.transaction.message.instructions;
+	const paymentInstruction = instructions[0];
+	const mainInstruction = instructions[instructions.length - 1];
+
+	const confirmation = await getConfirmation(
+		solanaContext,
+		parsedTransaction.transaction.signatures[0],
+	);
+
+	const status =
+		confirmation === 'finalized' || confirmation === 'confirmed'
+			? 'success'
+			: confirmation === 'processed'
+			? 'pending'
+			: 'failed';
```

Comment:
does transaction query from signature include commitment status, making a new request to just get status might be not efficient if It already existed